### ISSUE TITLE
Events

### ISSUE THREAD

**Issue by rickardliljeberg** [OdeToCode/Memflex#42]
My multi tenant solution is now online and seems to work but I will let it run for a bit longer before i create a pull request.

However, one thing that bugs me is that Memflex calles context save on it's own, making it hard to create proper transactions.

One solution would be events for actions before and after save, allowing other code to react to memflex's doings

**Comment by OdeToCode** [comment]
Yes, that was a decision I debated for a long time. In the end I figured the ability to use a new System.Transaction to make a transaction would be something people could use, and probably safer than someone forgetting to commit or call save. But, I don't know - it could certainly be something to change.
